praktikumsplaner-frontend/frontend/src/components/ExcelImportNWK.vue

```diff
@@ -1,97 +1,142 @@
+                        @click="handleFileImport()"
+                        >{{ buttonActionText }}
                     </v-btn>
+                    <input
+                        ref="uploader"
+                        class="d-none"
+                        type="file"
+                        :accept="excelFormat"
+                        @change="onFileChanged"
+                    />
                 </v-card-actions>
             </v-card>
         </v-form>
-    </v-dialog>
+        <v-btn
+            outlined
+            text
+            color="primary"
+            class="buttonEnd"
+            @click="cancel()"
+        >
+            Zurück
+        </v-btn>
+    </v-col>
 </template>
 
 <script setup lang="ts">
-import { ref } from "vue";
-import { useSnackbarStore } from "@/stores/snackbar";
-import { Levels } from "@/api/error";
+import { onMounted, ref } from "vue";
 import ExcelService from "@/api/ExcelService";
-import { useRules } from "@/composables/rules";
+import { EventBus } from "@/EventBus";
+import router from "@/router";
 
-const visible = ref<boolean>();
 const excelDatei = ref<File>();
-const form = ref<HTMLFormElement>();
-const snackbarStore = useSnackbarStore();
+const uploader = ref<HTMLInputElement>();
 const excelFormat =
     "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
-const validationRules = useRules();
-const rules = [
-    validationRules.fileTypeRule(
-        excelFormat,
-        "Es werden nur Excel Dateien akzeptiert"
-    ),
-];
+const buttonActionText = ref<string>("Auswählen");
+const isInUploadStage4 = ref<boolean>(false);
+const isInUploadedStage = ref<boolean>(false);
+const iconUpload = ref<string>("mdi-file-alert");
+const textUpload = ref<string>("Wählen Sie die gewünschte Exeldatei aus!");
+
+onMounted(() => {
+    EventBus.$emit("changeAppHeader", "Excel Datei hochladen");
+});
 
 function cancel() {
-    visible.value = false;
-    form.value?.reset();
+    router.push("/");
+}
+
+function handleFileImport() {
+    if (isInUploadStage4.value == true) {
+        uploadFile();
+    } else if (isInUploadedStage.value == true) {
+        router.push("/");
+    } else {
+        // Trigger click on the FileInput
+        uploader.value?.click();
+    }
+}
+
+function onFileChanged(e: any) {
+    excelDatei.value = e.target.files[0];
+    buttonActionText.value = "Hochladen";
+    isInUploadStage4.value = true;
+}
+
+function onUploadedSuccess() {
+    iconUpload.value = "mdi-paperclip-check";
+    textUpload.value = "Ihre Exceldatei wurde erfolgreich Hochgeladen!";
+    buttonActionText.value = "fortfahren";
+}
+
+function onUploadedFailed() {
+    iconUpload.value = "mdi-paperclip-remove";
+    textUpload.value = "Ihre Exceldatei konnte nicht hochgeladen werden!";
+    buttonActionText.value = "fortfahren";
+    let name = document.getElementById("card") as HTMLElement;
+    name?.style.setProperty("background-color", "rgba(216, 47, 67, 0.50)");
```

Comment:
Maybe trigger this via css-classes and class bindings. I think this would be a bit  cleaner.

```
<v-card id="card"  tile :class="{ 'uploadError': hasError }">
</v-card>

<style scoped>
.uploadError {
background-color: 
rgba(216, 47, 67, 0.50)
}
</style>
```